### PR TITLE
action: use the host for running the coverage

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use the host for running the coverage so there is no need to run DinD

## Why is it important?

ITs are unstable when running DinD, so let's avoid using that

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -40,7 +40,14 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - run: make docker-coverage
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
+        cache: 'pip'          # caching pip dependencies
+
+    - run: make all-requirements
```
Comment (v1v) [inline .github/workflows/test.yml:48]:
```suggestion
    - run: make requirements-tests
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -40,7 +40,14 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - run: make docker-coverage
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
+        cache: 'pip'          # caching pip dependencies
+
+    - run: make requirements-tests
```
Comment (v1v) [inline .github/workflows/test.yml:48]:
```suggestion
    - run: make requirements
    - run: make requirements-tests
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -40,7 +40,15 @@ jobs:
...
     - uses: actions/checkout@v3
 
-    - run: make docker-coverage
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
+        cache: 'pip'          # caching pip dependencies
+
+    - run: make requirements
+    - run: make requirements-tests
```
Comment (aspacca) [inline .github/workflows/test.yml:49]: please. execute `make requirements-tests` before `make requirements`: it's not optimal but this way we are sure that any sub-dependencies version from `requirements-tests` will not be "skewed" from how it should be from `requirements`

Comment (aspacca) [reply .github/workflows/test.yml:49]: btw, looking at the errors in the last run I have the impression that every `run` directive is executed in a different "context".
maybe the following?
```
run: make requirements-test && make requirements && make coverage
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -40,7 +40,15 @@ jobs:
...
+
+    - run: make coverage
```
Comment (v1v) [inline .github/workflows/test.yml:51]:
```suggestion
    - run: make requirements-test && make requirements && make coverage
```

Comment (aspacca) [reply .github/workflows/test.yml]: sorry, there's a typo: `requirements-tests`

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -40,7 +40,12 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - run: make docker-coverage
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
+        cache: 'pip'          # caching pip dependencies
+
+    - run: make requirements-test && make requirements && make coverage
```
Comment (v1v) [inline .github/workflows/test.yml:48]:
```suggestion
    - run: make requirements-tests && make requirements && make coverage
```

File: .github/workflows/test.yml (hunk 3 above)
Comment (v1v) [reply .github/workflows/test.yml]: As far as I see in the examples for using the [`setup-python`](https://github.com/actions/setup-python#caching-packages-dependencies)

```
steps:
- uses: actions/checkout@v3
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'pip'
    cache-dependency-path: '**/requirements-dev.txt'
- run: pip install -r subdirectory/requirements-dev.txt
```

Maybe the issue could be related to `pip3` used within the `make` goals

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -40,7 +40,12 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - run: make docker-coverage
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
+        cache: 'pip'          # caching pip dependencies
+
+    - run: make requirements-tests && make requirements && make coverage
```
Comment (v1v) [inline .github/workflows/test.yml:48]:
```suggestion
    - run: pip install -r requirements-tests.txt
    - run: pip install -r requirements.txt
    - run: make coverage
```

**Comment by aspacca** [comment]
we need to set dummy env variables:
```
AWS_ACCESS_KEY_ID=AWS_ACCESS_KEY_ID
AWS_SECRET_ACCESS_KEY=AWS_SECRET_ACCESS_KEY
```

**Comment by v1v** [comment]
> we need to set dummy env variables:

Do you mean within the make build system rather than the GitHub actions?

**Comment by aspacca** [comment]
> Do you mean within the make build system rather than the GitHub actions?

in the GitHub actions only, it does not seem they are set or that the `run: make coverage` sees them.
the aws python client connects to localstack running on docker, but it expects some credentials. that we provide through the two above environment variables. they can have dummy value since localstack does not really care for them

**Comment by aspacca** [comment]
>ModuleNotFoundError: No module named 'unittest.mock'
that's very unlikely: it is a builtin module

**Review by aspacca — APPROVED** [review]

**Pull request merged**